src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -240,13 +257,6 @@ inline oop ShenandoahHeap::evacuate_object(oop p, Thread* thread) {
     } else if (mark.age() >= InitialTenuringThreshold) {
       oop result = try_evacuate_object(p, thread, r, OLD_GENERATION);
       if (result != NULL) {
-        // TODO: Just marking the cards covering this object dirty
-        // may overall be less efficient than scanning it now for references to young gen
-        // or other alternatives like deferred card marking or scanning.
-        // We should revisit this.
-        // Furthermore, the object start should be registered for remset scanning.
-        MemRegion mr(cast_from_oop<HeapWord*>(result), result->size());
```

Comment:
Don't we need this? The object going into the old region may have pointers to young that need to be scanned.

Reply:
We do need this, but not necessarily right here. I moved this close to where the object transitions, in try_evacuate_object. When a thread succeeds in moving the object there, its card is dirtied and it is registered right away. See line ~340, where it says "// Successfully evacuated. Our copy is now the public one!".